src/util/Path.cpp

```diff
@@ -23,214 +17,153 @@ Path::Path(const char* path)
 {
 }
 
-Path::~Path()
+Path& Path::operator=(string p)
 {
+	this->path = std::move(p);
+	return *this;
 }
 
-/**
- * @return true if empty
- */
-bool Path::isEmpty()
+Path& Path::operator=(const char* p)
 {
-	return path.empty();
+	this->path = p;
```

Comment:
This string should be copied because `p` could be freed outside of `Path`.

Reply:
It is copied. this->path is a std::string, passing a char* to it will copy the C-String.

Reply:
```suggestion
	this->path = strdup(p);
```

Reply:
Your suggestion will lead to an memory leak.

Reply:
Nevermind, I forgot that `path` is a `std::string` and not a `char*`.